nixos/modules/system/boot/loader/systemd-boot/systemd-boot.nix

```diff
@@ -241,6 +241,20 @@ in {
       '';
     };
 
+    rememberLastChoice = mkOption {
+      default = false;
+
+      type = types.bool;
+
+      description = lib.mdDoc ''
+        Remembers the last chosen systemd-boot entry. For example, given entries A and B. If B is
+	is selected, next boot B will automatically be highlighted instead of the latest generation.
+	This is done by setting `default @saved` within the systemd-boot config.
+
+	This option will not work properly unless `boot.loader.canTouchEfiVariables = true`
```

Comment:
As noted this requires other flags to operate properly. It seems possible to enforce this in the assertions below, but I am unfamiliar with the syntax. Are there simpler examples and/or additional resources to look at?

Reply:
The assertions are just another module defined in `nixos/modules/misc/assertions.nix`; the `message` is thrown if the `assertion` value is false. Something like this should work:
```
      {
        assertion = config.boot.loader.systemd-boot.rememberLastChoice -> config.boot.loader.efi.canTouchEfiVariables;
        message = "systemd-boot rememberLastChoice requires mutable EFI variables.";
      }
```
`->` is just the logical implication operator and is often used in assertions. It's the same as `!a || b` i.e only enforce that b is true if a is also true. It just makes the intended behavior more obvious.